test/services/keyVault/keyVault-secret-tests.js

```diff
@@ -0,0 +1,377 @@
+var KvUtils = require('./kv-test-utils.js');
+var util = require('util');
+
+var series = KvUtils.series;
+var assertExactly = KvUtils.assertExactly;
+var compareObjects = KvUtils.compareObjects;
+var validateSecretBundle = KvUtils.validateSecretBundle;
+var validateSecretList = KvUtils.validateSecretList;
+var random = KvUtils.getRandom();
+
+var vaultUri = process.env['AZURE_KV_VAULT'];
+if (!vaultUri) {
+    vaultUri = 'https://nodesdktest.vault.azure.net';
+}
+
+var SECRET_NAME = 'nodeSecret';
+var SECRET_VALUE = 'Pa$$w0rd';
+var LIST_TEST_SIZE = 5; // TODO: down from 50 to avoid test timeout.
+
+describe('Key Vault secrets', function () {
+
+  var client;
+  var suiteUtil;
+
+  before(function (done) {
+    var credentials = new KeyVault.KeyVaultCredentials(KvUtils.authenticator);
+    client = new KeyVault.KeyVaultClient(credentials);
+      
+    suiteUtil = new MockedTestUtils(client, 'keyVault-secret-tests');
+    suiteUtil.setupSuite(done);
+  });
+
+  after(function (done) {
+    suiteUtil.teardownSuite(done);
+  });
+
+  beforeEach(function (done) {
+    suiteUtil.setupTest(done);
+  });
+
+  afterEach(function (done) {
+    suiteUtil.baseTeardownTest(done);
+  });
+  
+  describe('identifier', function() {
+    it('should work', function(done) {
+      
+      function assertMatch(vault, name, version, secretId) {
+        assertExactly(util.format('%s/secrets/%s', vault, name), secretId.baseIdentifier);
+        if (version) {
+          assertExactly(util.format('%s/secrets/%s/%s', vault, name, version), secretId.identifier);
+        } else {
+          assertExactly(secretId.baseIdentifier, secretId.identifier);
+        }
+        assertExactly(vault, secretId.vault);
+        assertExactly(name, secretId.name);
+        assertExactly(version, secretId.version);
+      }
+
+      function verifyCreate(vault, name, version) {
+        var secretId, parsedId;
+        if (version) {
+          secretId = KeyVault.createSecretIdentifier(vault, name, version);
+        } else {
+          secretId = KeyVault.createSecretIdentifier(vault, name);
+        }
+        assertMatch(vault, name, version, secretId);
+        if (version) {
+          parsedId = KeyVault.parseSecretIdentifier(secretId.identifier);
+          assertMatch(vault, name, version, parsedId);
+        }
+        parsedId = KeyVault.parseSecretIdentifier(secretId.baseIdentifier);
+        assertMatch(vault, name, null, parsedId);
+      }
+
+      verifyCreate(vaultUri, SECRET_NAME, null);
+      verifyCreate(vaultUri, SECRET_NAME, '1234');
+      
+      done();
+      
+    });
+  });
+  
+  describe('CRUD operations', function () {
+    it('should work', function (done) {
+      
+      var createdBundle;
+      var secretId;
+
+      function createSecret(next) {
+        client.setSecret(vaultUri, SECRET_NAME, { value: SECRET_VALUE }, function(err, secretBundle) {
+          if (err) throw err;
```

Comment:
do you have to throw, or just give the error to callback which is recommended, like "return next(err)". I believe the "next" should be the 'callback', right?